src/Components/MatchHistoryComp/MatchHistoryComp.js

```diff
@@ -0,0 +1,59 @@
+import React from 'react';
+import axios from 'axios';
+import { getFullUrl , findChampions } from '../../Modules/Helper';
+import { Spinner } from 'react-bootstrap';
+
+class MatchHistoryComp extends React.Component {
+
+    state = {
+        match_history: null
+    }
+
+    componentDidUpdate() {
+        if (this.props.display) {
+            console.log("Updated");
+            if (!this.state.match_history) {
+                console.log("Going to ajax");
+                let account_id = this.props.id;
+                let history_url = getFullUrl("https://eun1.api.riotgames.com/lol/match/v4/matchlists/by-account/" + account_id);
+                history_url += "&endIndex=10";
+                axios.get(history_url).then(res => {
+                    console.log(res);
+                    this.setState({
+                        match_history: res.data.matches
+                    });
+                })
+            }
+        }
+    }
+
+
+    render () { 
+        if (!this.props.display) {
+            return null;
+        }
+
+        let displayed_match_history = (
```

Comment:
Let's add a state to the component, called "isLoading", and while it does, we display a spinner. If it doesn't, show the data.